### ISSUE TITLE
Signedness Checker emits false positive when Object is cast to a Character and passed into method with parameter of type float or double

### ISSUE THREAD

**Issue by DmitriyShepelev** [typetools/checker-framework#3711]
The Signedness Checker emits a false positive when an `Object` is cast to a `Character` and passed into a method with a parameter of type `float` or `double`:
```
public class CharacterCast {
    void castCharacter(Object o) {
        floatParameter((Character) o);
        doubleParameter((Character) o);
    }

    void floatParameter(float f) {}

    void doubleParameter(double d) {}
}
```
Output:
```
root@DESKTOP-MO4BCEL:~/checker-framework# javacheck -processor SignednessChecker ~/temp-checker/checker-framework/checker/tests/signedness/CharacterCast.java
/root/temp-checker/checker-framework/checker/tests/signedness/CharacterCast.java:3: error: [argument.type.incompatible] incompatible argument for parameter f of floatParameter.
        floatParameter((Character) o);
                       ^
  found   : @Unsigned Character
  required: @Signed float
/root/temp-checker/checker-framework/checker/tests/signedness/CharacterCast.java:4: error: [argument.type.incompatible] incompatible argument for parameter d of doubleParameter.
        doubleParameter((Character) o);
                        ^
  found   : @Unsigned Character
  required: @Signed double
2 errors
```
`javacheck` alias:
`alias javacheck='$CHECKERFRAMEWORK/checker/bin/javac'`

What can be changed in the Checker Framework to fix this?
